driver-core/src/main/com/mongodb/internal/connection/Authenticator.java

```diff
@@ -105,4 +105,9 @@ public void reauthenticate(final InternalConnection connection) {
                 "Reauthentication requested by server but is not supported by specified mechanism.");
     }
 
+    public void reauthenticateAsync(final InternalConnection connection, final SingleResultCallback<Void> callback) {
+        throw new UnsupportedOperationException(
+                "Reauthentication requested by server but is not supported by specified mechanism.");
+    }
```

Comment:
It seems that now this method should call `authenticateAsync`.

Reply:
I think it should be called reauth, because it is what is done in response to a 391, that is, reauthenticate. This is not the same as what should be done to authenticate for all authenticators.

Reply:
> I think it should be called reauth

I did not suggest renaming the method.